### PR TITLE
CORE-101 don't attempt to create shell google project if key exists

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: https://broadworkbench.atlassian.net/browse/CORE-101

  \<Don't forget to include the ticket number in the PR title!\>

What:

  \<For your reviewers' sake, please describe in a sentence or two what this PR is accomplishing (usually from the users' perspective, but not necessarily).\>

Why:

  \<For your reviewers' sake, please describe in ~1 paragraph what the value of this PR is to our users or to ourselves.\>

How:

  \<For your reviewers' sake, please describe in ~1 paragraph how this PR accomplishes its goal.\>

  \<If the PR is big, please indicate where a reviewer should start reading it (i.e. which file or function).\>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 1]
```diff
@@ -46,6 +45,10 @@ import scala.jdk.CollectionConverters._
 object GoogleExtensions {
   val resourceId = ResourceId("google")
   val getPetPrivateKeyAction = ResourceAction("get_pet_private_key")
+
+  def getShellGoogleProjectName(user: SamUser, googleServicesConfig: GoogleServicesConfig) =
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala:49]: moved to be in a more accessible spot for tests

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 2]
```diff
@@ -485,9 +488,6 @@ class GoogleExtensions(
       }
     } yield key
 
-  def getArbitraryPetServiceAccountKey(user: SamUser, samRequestContext: SamRequestContext): IO[String] =
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala:488]: this was inlined

**Review by calypsomatic — APPROVED** [review]

**Review by davidangb — APPROVED** [review]
:+1:, with a question: this is still going to be vulnerable to multiple requests hitting `getArbitraryPetServiceAccountKey` in short order - if the user has multiple tabs open, or even if a script or the UI shoots off multiple requests that each involve a pet.

Certainly this PR is better than before, so it's worth merging; is it worth protecting against multiple requests?

**Review by davidangb — APPROVED** [review]
+1 again, and thank you!

**Pull request merged**